IQSS/dataverse-frontend
# 116 - Publish npm package for the design system

## What this PR does / why we need it:
This PR publishes the Dataverse Design System to the npm registry. It includes the necessary changes and configurations to publish the design system components and make them available as an npm package.

## Which issue(s) this PR closes:
Closes #116 

## Special notes for your reviewer:
The automation of the publishing will be developed in this other issue:

- #140

The creation of a test registry will be developed in this other issue:

- #144

To publish a new version of the package you need to be registered in the npmjs.com IQSS organization. Contact me via slack so I can send you the invite!

## Suggestions on how to test this:

### Test the published package

To test the package:

1. Create a [new React application](https://react.dev/learn/start-a-new-react-project). I'll go with the first option which is running `npx create-next-app` in your machine (ensure you have node installed in your machine)
7. In your app root folder run `npm i` to install all the packages of the demo app
 8. Follow the instructions from the README of the package [@iqss/dataverse-design-system](https://www.npmjs.com/package/@iqss/dataverse-design-system/v/1.0.1)
9. If you're using `npx create-next-app` then go to the `src/app/page.tsx` file
 10. Place the imports at the top of the file with the other imports `import { Col, Container, Form, Row} from "@iqss/dataverse-design-system";`. If you are using next-js you'll need to set `'use client'` at the top of the file since the design system components are client components
 11.  Now copy the example code in the usage section of the package README and paste it somewhere inside the  Home function of the `src/app/page.tsx` file
 12. Run the application using `npm run dev` and go to http://localhost:3000 to see the application using the design system components

 
## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
None

## Is there a release notes update needed for this change?:
Published first version of the Dataverse Design System to the npm registry [@iqss/dataverse-design-system](https://www.npmjs.com/package/@iqss/dataverse-design-system/v/1.0.1) 

## Additional documentation:
[Lerna publishing](https://lerna.js.org/docs/features/version-and-publish)
